providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/options/GoogleComputeEngineTemplateOptions.java

```diff
@@ -18,266 +18,393 @@
+	public GoogleComputeEngineTemplateOptions tags(Iterable<String> tags) {
+		return GoogleComputeEngineTemplateOptions.class.cast(super.tags(tags));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public GoogleComputeEngineTemplateOptions wrapInInitScript(
+			boolean wrapInInitScript) {
+		return GoogleComputeEngineTemplateOptions.class.cast(super
+				.wrapInInitScript(wrapInInitScript));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public GoogleComputeEngineTemplateOptions runScript(String script) {
+		return GoogleComputeEngineTemplateOptions.class.cast(super
+				.runScript(script));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public GoogleComputeEngineTemplateOptions blockOnComplete(
+			boolean blockOnComplete) {
+		return GoogleComputeEngineTemplateOptions.class.cast(super
+				.blockOnComplete(blockOnComplete));
+	}
+
+	public boolean canIpForward() {
+		return canIpForward;
+	}
+
+	/**
+	 * @ee https://cloud.google.com/compute/docs/networking#canipforward
+	 * @return
+	 */
+	public void canIpForward(boolean enable) {
+		this.canIpForward = enable;
+	}
+
+	/**
+	 * @see #getDisks()
+	 * @see org.jclouds.googlecomputeengine.domain.templates.InstanceTemplate.AttachDisk
+	 */
+	public GoogleComputeEngineTemplateOptions addDisk(AttachDisk disk) {
+		this.disks.add(disk);
+		return this;
+	}
+
+	public void autoCreateDisk(final AutoCreateDiskOptions diskOptions) {
+		this.autoCreateDiskOptions = diskOptions;
+	}
+
+	public AutoCreateDiskOptions getAutoCreateDiskOptions() {
+		return autoCreateDiskOptions;
+	}
+
+	/**
+	 * @return the AttachDisks for this instance.
+	 */
+	public List<AttachDisk> getDisks() {
+		return disks;
+	}
+
+	/**
+	 * @see #getDisks()
+	 * @see org.jclouds.googlecomputeengine.domain.templates.InstanceTemplate.AttachDisk
+	 */
+	public GoogleComputeEngineTemplateOptions disks(List<AttachDisk> disks) {
+		this.disks = Lists.newArrayList(disks);
+		return this;
+	}
+
+	public static class AutoCreateDiskOptions {
+		public final AttachDisk.Type diskType;
+		public final AttachDisk.Mode diskMode;
+		public final boolean isBootDisk;
+		public final int diskSizeGb;
+		private String diskName = null;
+
+		public AutoCreateDiskOptions(final AttachDisk.Type diskType,
+				final AttachDisk.Mode diskMode, final boolean isBootDisk,
+				final int diskSizeGb) {
+			this.diskType = diskType;
+			this.diskMode = diskMode;
+			this.isBootDisk = isBootDisk;
+			this.diskSizeGb = diskSizeGb;
+		}
+
+		public AutoCreateDiskOptions(final AttachDisk.Type diskType,
+				final AttachDisk.Mode diskMode, final boolean isBootDisk,
+				final int diskSizeGb, final String diskName) {
+			this(diskType, diskMode, isBootDisk, diskSizeGb);
+			// TODO: Validate against regex according to GCE spec. What is the GCE spec?
+			this.diskName = diskName;
+		}
+
+		public String getDiskName(final String nodeName) {
+			if (null != diskName)
+				return diskName;
+			else
+				return "jclouds-" + UUID.randomUUID().toString();
+		}
+	}
```

Comment:
This class has been completely reformatted. Please, keep the formatting it had (basically a 3 space indent, but have a look at the [jclouds code style](https://cwiki.apache.org/confluence/display/JCLOUDS/Coding+Standards)) so the only changes in it are the lines relevant to your patch.